server/container/util-java8/src/main/java/org/apache/james/util/CompletableFutureUtil.java

```diff
@@ -26,6 +26,54 @@
 
 public class CompletableFutureUtil {
 
+    public static <T> CompletableFutureOfStream<T> chainOnStream(CompletableFuture<Stream<T>> streamCompletableFuture) {
+        return CompletableFutureOfStream.of(streamCompletableFuture);
+    }
+
+    public static class CompletableFutureOfStream<T> {
+        private final CompletableFuture<Stream<T>> completableFuture;
+
+        public static <T> CompletableFutureOfStream<T> of(CompletableFuture<Stream<T>> completableFuture) {
+            return new CompletableFutureOfStream(completableFuture);
+        }
+
+        private CompletableFutureOfStream(CompletableFuture<Stream<T>> completableFuture) {
+            this.completableFuture = completableFuture;
+        }
+
+        public CompletableFutureOfStream<T> performOnAll(Function<T, CompletableFuture<Void>> action) {
+            return CompletableFutureOfStream.of(
+                CompletableFutureUtil.performOnAll(completableFuture(), action));
+        }
+
+        public <U> CompletableFutureOfStream<U> map(Function<T, U> function) {
+            return CompletableFutureOfStream.of(
+                CompletableFutureUtil.map(completableFuture(), function));
+        }
+
+        public <U> CompletableFutureOfStream<U> thenComposeOnAll(Function<T, CompletableFuture<U>> function) {
+            return CompletableFutureOfStream.of(
+                CompletableFutureUtil.thenComposeOnAll(completableFuture(), function));
+        }
+
+        public <U> CompletableFutureOfStream<U> flatMap(Function<T, Stream<U>> function) {
+            return CompletableFutureOfStream.of(completableFuture().thenApply(stream ->
+                stream.flatMap(function)));
+        }
+
+        public <U> CompletableFutureOfStream<U> thenCompose(Function<Stream<T>, CompletableFuture<Stream<U>>> function) {
+            return CompletableFutureOfStream.of(completableFuture().thenCompose(function));
+        }
+
+        public CompletableFuture<Stream<T>> completableFuture() {
+            return this.completableFuture;
+        }
+
+        public Stream<T> join() {
```

Comment:
I think it cost nothing to add and that if we use it often, it's nice to have it.